### PR TITLE
Specify background attributionsrc requests

### PR DESCRIPTION
**Pull request by apasel422**
***
<a href="https://pr-preview.s3.amazonaws.com/apasel422/attribution-reporting-api/pull/723.html" title="Last updated on Mar 29, 2023, 3:28 PM UTC (c89bc83)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/723/4e0d46a...apasel422:c89bc83.html" title="Last updated on Mar 29, 2023, 3:28 PM UTC (c89bc83)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -785,6 +805,95 @@ and a [=suitable origin=] |reportingEndpoint|:
...
+1. Let |triggerHeader| be the result of [=header list/get|getting=]
+    "`Attribution-Reporting-Register-Trigger`" from |response|'s
+    [=response/header list=].
+1. If |eligibility| is:
+    <dl class="switch">
+    : "<code>[=eligibility/navigation-source=]</code>"
+    :: Run the following steps:
+        1. If |sourceHeader| is null, return.
+        1. Let |source| be the result of running
+            [=parse source-registration JSON=] with |sourceHeader|,
```
[the remark this answers is not in the document]
Comment (apasel422) [reply index.bs:865]: Agreed that this is an error in my PR, but do you also think it's an error in the implementation?

https://source.chromium.org/chromium/chromium/src/+/main:net/http/http_response_headers.h;l=155-158;drc=0c65172f8db94dfbf734a9d31234bc2061e1211f

Not sure what the corresponding caveats are in blink, if any.

[the remark this answers is not in the document]
Comment (csharrison) [reply index.bs:865]: Hm, in the implementation I suppose we should treat the header values as arbitrary bytes and fail parsing if not UTF-8. Here i think we can use https://encoding.spec.whatwg.org/#utf-8-decode-without-bom-or-fail

**Comment by csharrison** [comment]
What you have is good but let's make sure our implementation matches it. In particular, this uses the "replacement" mode of UTF-8 decoding which replaces errors with "�"

**Comment by apasel422** [comment]
> What you have is good but let's make sure our implementation matches it. In particular, this uses the "replacement" mode of UTF-8 decoding which replaces errors with "�"

IIUC most headers essentially only support ASCII; not sure we want to break from that.

**Pull request merged**